### DISCUSSION TITLE
Wert des Analog Meter nicht immer in Raw Value

### DISCUSSION THREAD

**Discussion by superhannsi** [jomjol/AI-on-the-edge-device#1777] [Q&A] [answered]
Guten Tag,

zunächst einmal möchte ich mich für das tolle Projekt bedanken.
Ich nutze es für einen anderen Anwendungsfall. Ich ermittle damit die Kesseltemperatur (Digital Counter) und den Status des Brenners (Analog Meter). Bislang funktionierte dies auch sehr gut.
Seit kurzem (vermutlich seit v13.0.8) habe ich das Problem, dass der Wert des Analog Meter nicht mehr zuverlässig im Raw Value enthalten ist. Der Analog Meter steht z.B. auf 1.0 aber der Raw Value dazu ist .0 anstatt .1
Gab es an dieser Stelle eine Änderung in o.g. Version, auf die ich jetzt innerhalb der Konfiguration Einluss nehmen kann?

**Comment by jomjol** [comment, marked as answer]
Was genau meinst du? Raw Value "0.0" statt "1.0"?

Wenn ja, dann hätte das eigentlich schon immer so sein sollen ;-). Der Ausgabewert des neuronalen Netzes sollte von 0.0 (=12 Uhr) über 5.0 (=6 Uhr) bis 9.9 wieder kurz vor 12 Uhr gehen. bei 10.0 (=1.0) sollte dann wieder 0.0 kommen.

**Reply by superhannsi** [reply]
Vielen Dank für die schnelle Antwort:
Ich habe das analoge Model so trainiert, dass es nur noch die Werte 0.0 bzw. 1.0 für den Analog Meter zurückgibt:
![grafik](https://user-images.githubusercontent.com/100438602/211139619-75894539-c543-4767-ab72-0a290b2cb6e1.png)

Ich dachte, dass dann dieser Wert als Nachkommastelle an den den Digital Counter angehängt wird. Mit dieser Nachkommastelle wollte ich im weiteren den Zustand als aus bzw. an interpretieren.
Bei diesem Beispiel:
![grafik](https://user-images.githubusercontent.com/100438602/211139796-07181da4-5005-4a72-a0eb-b7787344b628.png)

hätte ich als Raw Value 74.1 erwartet.
![grafik](https://user-images.githubusercontent.com/100438602/211139915-50c87006-d0d9-461e-b002-37dd0b81d00b.png)

Sollte ich zwei andere Ausgabewerte für den Analog Meter verwenden, um bei jedem Durchlauf eine evtl. Zustandsänderung erkennen zu können oder kann ich hier etwas über die Konfiguration beeinflussen?

**Reply by ghost** [reply]
Hast du vielleicht CheckDigitIncreaseConsistency = true? Das könnte daran liegen. Bin mir aber nicht sicher.

**Reply by superhannsi** [reply]
Vielen Dank für den Hinweis. Steht bei mir allerdings auf false.
Nachdem ich mit den Änderungen innerhalb der Konfiguration nicht weiterkam, hab ich es nun anders hinbekommen.
Ich habe das Model für den Analog Meter erneut trainiert. Zuvor hatte ich den Zustand "aus" als 0.0 definiert. Dies habe ich nun in 9.9 geändert. Ich habe das nun heute den Tag über laufen lassen und gerade nochmal die Werte aus dem Data Graph mit den geloggten Bildern des Analog Meter verglichen. Das sieht nun ziemlich gut aus. Ich hab in ca. 98% der Fälle das gewünschte Ergebnis.

**Reply by ghost** [reply]
9.8-0.2 hat Sonderbehandlungen. Ich würde eher auf 3.0 und 7.0 trainieren. Für diese Werte gibt es keinen Postprozess

**Reply by superhannsi** [reply]
Vielen Dank. Es läuft mit den aktuellen Einstellungen relativ stabil. Falls ich doch nochmal eingreifen muss, würde ich es mit den von Dir vorsgeschlagenen Werten versuchen.
Ich habe mich jetzt doch dazu entschlossen das Monitoring der Heizungsdaten etwas größer anzugehen.
Daher wird die ESP32CAM demnächst einen anderen Platz finden und muss dann neu trainiert werden.

**Comment by sammykd** [comment]
Hallo superhannsi,
gehört nicht hier zu deiner Lösung aber evtl. Anregung. So wie das aussieht, handelt es sich hier um eine Buderus Anlage oder liege ich da falsch?
Vieleicht mal eine Anregung/ TIP für dich. Ich habe eine Buderus GB162 mit **Internet-Modul KM50**, die Werte die Du erfasst, kann man über IO-Broker mittels der Instanz EMS.ESP parallel zur normalen Buderus APP erfassen und viele weitere Werte. Voraussetzung ist aber das Internetmodul für die Heizung bei mir ein KM50 + IOBROKER für weiter Funktionen mit Homematic und dann zur Visualisierung noch INFLUXDB und Grafana, das sieht dann in etwa so aus. Vielleicht ist das ja mal eine Anregung für dich. **GAS und Wasser erfasse ich mit dem Projekt von JOMJOL über MQTT an IOBROKER**.  Allerdings ist das dann doch schon mal ein Projekt in dem man viel Zeit investieren muss.
![gb162](https://user-images.githubusercontent.com/57226083/211189549-97cb9af5-7690-4d80-a5f5-184fd1759205.png)

**Reply by superhannsi** [reply]
Hallo sammykd,

nein, das hast Du völlig richtig erkannt. Es handelt sich dabei um eine GB142 aus dem Jahr 2004. Das Bild ist von dem Basiscontroller BC10, der sich direkt am Wandgerät befindet. Die Regeleinheit RC35 hängt bei mir im Wohnzimmer.
Vielen Dank für Deine Anregungen. Das Dashboard sieht toll aus.
Das KM50 ist für die doch schon recht alte Heizung wahrscheinlich etwas überdimensioniert. Ich bin am Wochenende zufällig auf das EMS Bus WIFI Gateway S32 gestoßen. Das liegt bei ca. 90€ und man kann es an die Servicebuchse anschließen. Dies sollte dann die auf Deinem Dashboard ersichtlichen Werte liefern.

Ich wünsche Dir noch ein schönes Wochenende!
